src/llmq/quorums_dkgsessionhandler.cpp

```diff
@@ -145,6 +137,35 @@ void CDKGSessionHandler::ProcessMessage(CNode* pfrom, const std::string& strComm
     }
 }
 
+void CDKGSessionHandler::StartThread()
+{
+    auto threadName = [&]() -> auto {
+        switch (params.type) {
+            case Consensus::LLMQ_50_60:
+                return "q-phase-1";
+            case Consensus::LLMQ_400_60:
+                return "q-phase-2";
+            case Consensus::LLMQ_400_85:
+                return "q-phase-3";
+            case Consensus::LLMQ_TEST:
+                return "q-phase-100";
+            case Consensus::LLMQ_DEVNET:
+                return "q-phase-101";
+            default:
+                throw std::runtime_error("Tried to start a CDKGSessionHandler thread for LLMQ_NONE.");
```

Comment:
Is this caught / handled somewhere? This seems to be adding a runtime_error which could potentially be caused by something received by p2p. What's the point in not just returning `""` or `something else.

Reply:
Those threads are meant to be started only once at startup.